core/ledger/kvledger/channelinfo_provider.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package kvledger
+
+import (
+	cb "github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric/bccsp/factory"
+	"github.com/hyperledger/fabric/common/channelconfig"
+	"github.com/hyperledger/fabric/common/ledger/blkstorage"
+	"github.com/hyperledger/fabric/protoutil"
+)
+
+// ChannelInfoProvider implements ledger.ChannelInfoProvider interface
+type ChannelInfoProvider struct {
+	BlkStoreProvider *blkstorage.BlockStoreProvider
+}
+
+// GetAllMSPIDs retrieves the MSPIDs of application organizations in all the channel configurations,
+// including current and previous channel configurations.
+func (p *ChannelInfoProvider) GetAllMSPIDs(channelName string) ([]string, error) {
+	blockStore, err := p.BlkStoreProvider.Open(channelName)
+	if err != nil {
+		return nil, err
+	}
+	blockchainInfo, err := blockStore.GetBlockchainInfo()
+	if err != nil {
+		return nil, err
+	}
+
+	// Iterate over the config blocks to get all the channel configs, extract MSPIDs and add to mspidsMap
+	mspidsMap := map[string]string{}
+	blockNum := blockchainInfo.Height
+	for blockNum > 0 {
```

Comment:
@cendhu Thank you. The blockNum usage in this PR is consistent to mostRecentEntryBelow(...) in https://github.com/hyperledger/fabric/blob/master/core/ledger/confighistory/db_helper.go#L73, where it retrieves the block below "blockNum", i.e., "blockNum - 1". It is clear to me since the first blockNum is blockHeight and the method is mostRecentConfigBlockBelow. 

@manish-sethi, I'd like to get your opinion. I am ok with the above suggestion if we all agree using the actual blockNum (instead of entryBelow) is more clear. In this case, the method will be renamed to "mostRecentConfigBlockAtOrBelow".

Reply:
In the configHistory store, as these lines are present next to each other, it does not create any confusion.
```
	if blockNum == 0 {
		return nil, errors.New("blockNum should be greater than 0")
	}
	startKey := encodeCompositeKey(ns, key, blockNum-1)
```

> where it retrieves the block below "blockNum", i.e., "blockNum - 1". It is clear to me since the first blockNum is blockHeight

Mixed usage of block height and blockNum is confusing me. Not just `mostRecentConfigBlockBelow`, as I mentioned in the earlier comment, `for blockNum > 0` is confusing too. This confusion does not arise while reading the code in configHistory store.

Reply:
> In this case, the method will be renamed to "mostRecentConfigBlockAtOrBelow".

`mostRecentConfigBlockAsOf(blockNum)` can also be used as mentioned in https://github.com/hyperledger/fabric/pull/1352#discussion_r432917059

Reply:
Yes, I agree that assigning the height to a variable named blockNum could cause confusion. `mostRecentConfigBlockAsOf(blockNum)` seems good to me.